src/ansi.go

```diff
@@ -89,14 +89,16 @@ func matchOperatingSystemCommand(s string) int {
 	//                        ^ match starting here
 	//
 	i := 5 // prefix matched in nextAnsiEscapeSequence()
-	for ; i < len(s) && isPrint(s[i]); i++ {
-	}
-	if i < len(s) {
-		if s[i] == '\x07' {
-			return i + 1
-		}
-		if s[i] == '\x1b' && i < len(s)-1 && s[i+1] == '\\' {
-			return i + 2
+	for ; i < len(s); i++ {
+		c := s[i]
+		if !isPrint(c) {
+			if c == '\x07' {
+				return i + 1
+			}
+			if c == '\x1b' && i+1 < len(s) && s[i+1] == '\\' {
+				return i + 2
+			}
+			break
```

Comment:
Can you explain why this version is faster?

Reply:
Good catch, and reverted.

I added this speculatively (based on the perf improvements to `matchControlSequence()`) and so that both would follow the same pattern. I just did some testing that found this change has no impact of performance so it has been reverted.